### PR TITLE
[4] com_modules empty state

### PR DESCRIPTION
**Pull request by PhilETaylor**
Can display for Site and Administrator Modules (Only for completeness, as near impossible to replicate for Admin Modules Empty State) 

### Testing Instructions

Trash the three modules Joomla 4 installed when you installed Joomla. Empty the trash. See the Empty State screen. Use the buttons to create a new Site Module, see that it works as expected 

For Administrator Modules - code review only ;-) 

Screenshot 2021-04-28 at 22 52 32
Screenshot 2021-04-28 at 22 52 53

### CODE REVIEW

File: administrator/language/en-GB/com_modules.ini [hunk 1]
```diff
@@ -25,6 +25,10 @@ COM_MODULES_CONFIGURATION="Module: Options"
 COM_MODULES_CUSTOM_OUTPUT="Custom Output"
 COM_MODULES_CUSTOM_POSITION="Active Positions"
 COM_MODULES_DESELECT="Deselect"
+COM_MODULES_EMPTYSTATE_TITLE_SITE='No Site Modules have been created yet.'
+COM_MODULES_EMPTYSTATE_TITLE_ADMINISTRATOR='No Administrator Modules have been created yet.'
+COM_MODULES_EMPTYSTATE_CONTENT='Modules are lightweight and flexible extensions used for page rendering, use them to add small blocks of functionality to your page.'
+COM_MODULES_EMPTYSTATE_BUTTON_ADD='Add a Module'
```
Comment (Quy) [inline administrator/language/en-GB/com_modules.ini]:
```suggestion
COM_MODULES_EMPTYSTATE_BUTTON_ADD='Add a Module'
COM_MODULES_EMPTYSTATE_CONTENT='Modules are lightweight and flexible extensions used for page rendering. Use them to add small blocks of functionality to your page.'
COM_MODULES_EMPTYSTATE_TITLE_ADMINISTRATOR='No Administrator Modules have been created yet.'
COM_MODULES_EMPTYSTATE_TITLE_SITE='No Site Modules have been created yet.'
```

Comment (PhilETaylor) [reply administrator/language/en-GB/com_modules.ini]: I literally suck... how many more times am I going to forget to do this hahhahaha

File: administrator/components/com_modules/src/Model/ModulesModel.php [hunk 2]
```diff
@@ -456,4 +456,33 @@ protected function getListQuery()
...
+			->clear('values')
+			->clear('bounded')
+			->clear('limit')
+			->clear('order')
+			->clear('where')
+			->clear('where')
+			->select('count(*)');
+
+		$sql->where($this->_db->quoteName('a.client_id') . ' = :client_id')
+			->bind(':client_id', $clientId);
```
Comment (Quy) [inline administrator/components/com_modules/src/Model/ModulesModel.php]:
```suggestion
		$clientId = (int) $this->getState('client_id');

		$sql = $this->query
			->clear('select')
			->clear('values')
			->clear('bounded')
			->clear('limit')
			->clear('order')
			->clear('where')
			->clear('where')
			->select('count(*)');

		$sql->where($this->_db->quoteName('a.client_id') . ' = :client_id')
			->bind(':client_id', $clientId, ParameterType::INTEGER);
```

Comment (PhilETaylor) [reply administrator/components/com_modules/src/Model/ModulesModel.php]: Thanks.

File: administrator/language/en-GB/com_modules.ini [hunk 3]
```diff
@@ -25,6 +25,10 @@ COM_MODULES_CONFIGURATION="Module: Options"
 COM_MODULES_CUSTOM_OUTPUT="Custom Output"
 COM_MODULES_CUSTOM_POSITION="Active Positions"
 COM_MODULES_DESELECT="Deselect"
+COM_MODULES_EMPTYSTATE_BUTTON_ADD='Add a Module'
```
Comment (Quy) [inline administrator/language/en-GB/com_modules.ini]:
```suggestion
COM_MODULES_EMPTYSTATE_BUTTON_ADD='Add a module'
```

File: administrator/components/com_modules/src/View/Modules/HtmlView.php [hunk 4]
```diff
@@ -65,6 +65,14 @@ class HtmlView extends BaseHtmlView
 	 */
 	public $activeFilters;
 
+	/**
+	 * Is this view an Empty State
+	 *
+	 * @var  boolean
+	 * @since __DEPLOY_VERSION__
+	 */
+	public $isEmptyState = false;
```
Comment (Quy) [inline administrator/components/com_modules/src/View/Modules/HtmlView.php]:
```suggestion
	private $isEmptyState = false;
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 23b41873402ba9331450469d7f08b28e5d1bdf11<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33398">issues.joomla.org/tracker/joomla-cms/33398</a>.</sub>

**Comment by saumyasarkar11** [comment]
I have tested this item :white_check_mark: successfully on 23b41873402ba9331450469d7f08b28e5d1bdf11<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33398">issues.joomla.org/tracker/joomla-cms/33398</a>.</sub>

File: administrator/components/com_modules/src/View/Modules/HtmlView.php (hunk 4 above)
Comment (PhilETaylor) [reply administrator/components/com_modules/src/View/Modules/HtmlView.php]: I have worked out that it is phpStorm doing this when you use phpStorm to add the property - Ill keep an eye on that in future, thanks.

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33398">issues.joomla.org/tracker/joomla-cms/33398</a>.</sub>

**Comment by drmenzelit** [comment]
Thanks

**Pull request merged**